dbms/src/Storages/DeltaMerge/SSTFilesToBlockInputStream.cpp

```diff
@@ -130,12 +135,27 @@ void SSTFilesToBlockInputStream::readPrefix()
     process_keys.lock_cf_bytes = 0;
 }
 
-void SSTFilesToBlockInputStream::readSuffix()
+SSTFilesToBlockInputStream::~SSTFilesToBlockInputStream() = default;
+
+void SSTFilesToBlockInputStream::readPrefix() {}
+
+void SSTFilesToBlockInputStream::checkFinishedState(SSTReaderPtr & reader)
 {
     // There must be no data left when we write suffix
-    assert(!write_cf_reader || !write_cf_reader->remained());
-    assert(!default_cf_reader || !default_cf_reader->remained());
-    assert(!lock_cf_reader || !lock_cf_reader->remained());
+    if (!write_cf_reader)
+        return;
+    if (!write_cf_reader->remained())
+        return;
+    RUNTIME_CHECK(soft_limit.has_value());
```

Comment:
If neither the previous two cases match, then this must be a parallel-handling-single-big-region-snapshot case, then the `cur` cursor is definetely beyond end_limit